packages/yoastseo/spec/languageProcessing/researches/keywordCountSpec.js

```diff
@@ -25,221 +26,779 @@ const buildMorphologyMockResearcher = function( keyphraseForms ) {
-		keyphraseForms: [ [ "key phrase" ] ],
+		description: "can match keyphrase followed by RTL-specific punctuation marks",
+		paper: new Paper( "<p>الجيدة؟</p>", { keyword: "الجيدة" } ),
+		keyphraseForms: [ [ "الجيدة" ] ],
 		expectedCount: 1,
-		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>key phrase</yoastmark>.",
-			original: "A string with a key phrase." } ) ],
-		skip: true,
+		expectedMarkings: [
+			new Mark( {
+				original: "الجيدة؟",
+				marked: "<yoastmark class='yoast-text-mark'>الجيدة</yoastmark>؟",
+				position: { endOffset: 9, startOffset: 3 } }
+			),
+		],
+		skip: false,
 	},
+];
+
+describe.each( testCasesWithSpecialCharacters )( "Test for counting the keyword in a text containing special characters",
+	function( {
+		description,
+		paper,
+		keyphraseForms,
+		expectedCount,
+		expectedMarkings,
+		skip } ) {
+		const test = skip ? it.skip : it;
+
+		test( description, function() {
+			const mockResearcher = buildMorphologyMockResearcher( keyphraseForms );
+			buildTree( paper, mockResearcher );
+			const keyWordCountResult = keyphraseCount( paper, mockResearcher );
+			expect( keyWordCountResult.count ).toBe( expectedCount );
+			expect( keyWordCountResult.markings ).toEqual( expectedMarkings );
+		} );
+	} );
+
+const testCasesWithLocaleMapping = [
 	{
-		// eslint-disable-next-line max-len
-		description: "with exact matching, a multi word keyphrase should not me counted if the focus keyphrase has the same words in a different order",
-		paper: new Paper( "A string with a phrase key.", { keyword: "\"key phrase\"" } ),
-		keyphraseForms: [ [ "key phrase" ] ],
+		description: "counts a string of text with German diacritics and eszett as the keyword",
+		paper: new Paper( "<p>Waltz keepin auf mitz auf keepin äöüß weiner blitz deutsch spitzen.</p>", { keyword: "äöüß", locale: "de_DE" } ),
+		keyphraseForms: [ [ "äöüß" ] ],
+		expectedCount: 1,
+		expectedMarkings: [
+			new Mark( { marked: "Waltz keepin auf mitz auf keepin <yoastmark class='yoast-text-mark'>äöüß</yoastmark> weiner blitz deutsch spitzen.",
+				original: "Waltz keepin auf mitz auf keepin äöüß weiner blitz deutsch spitzen.",
+				position: { endOffset: 40, startOffset: 36 } } ) ],
+		skip: false,
+	},
+	{
+		description: "doesn't count a match if the keyphrase is with diacritic while the occurrence in the text is with diacritic",
+		paper: new Paper( "<p>Acción Española fue una revista.</p>", { keyword: "accion", locale: "es_ES" } ),
+		keyphraseForms: [ [ "accion" ] ],
 		expectedCount: 0,
 		expectedMarkings: [],
-		skip: true,
+		skip: false,
 	},
 	{
-		description: "with exact matching, it should match a full stop if it is part of the keyphrase and directly precedes the keyphrase.",
-		paper: new Paper( "A .sentence with a keyphrase.", { keyword: "\".sentence\"" } ),
-		keyphraseForms: [ [ ".sentence" ] ],
+		description: "counts a string of text with Spanish accented vowel",
```

Comment:
This is the same behavior as the current released version. I don't know if it's desirable or if it was a conscious decision. It's hard to always provide context for EVERY case in this PR because of the lack of documentation that we have about keyphrase matching in the first place, on top of the already huge scope of this PR.

This is not an excuse for sure :D just providing you with context on how I missed adding extra context for each of those cases :D 
I'll add the context where I can 👍🏽

Reply:
I added an additional comment in `matchWordFormInTokens` function inside `packages/yoastseo/src/languageProcessing/helpers/match/matchWordFormsWithSentence.js` file, and also add in the spec file itself :)